cdec/installation-manager-core/src/main/java/com/codenvy/im/artifacts/helper/CDECMultiServerHelper.java

```diff
@@ -299,9 +299,13 @@ public Command getUpdateCommand(Version versionToUpdate, Path pathToBinaries, In
 
         switch (step) {
             case 0:
-                return createCommand(format("rm -rf %1$s; " +
-                                            "mkdir %1$s; " +
-                                            "unzip -o %2$s -d %1$s", getTmpCodenvyDir(), pathToBinaries.toString()));
+                return new MacroCommand(ImmutableList.of(
+                    createCommand(format("rm -rf %1$s; " +
+                                         "mkdir %1$s; " +
+                                         "unzip -o %2$s -d %1$s; ", getTmpCodenvyDir(), pathToBinaries.toString())),
+                    createCommand("sudo systemctl stop crond"),
+                    createCommand("sudo systemctl stop puppet"),
```

Comment:
any migration must be performed before update tomcat, also we must stop current codenvy server to protect DB from unexpected changes, so we need to stop puppet because it will restart codenvy while we running migrations.